youtube_dl/extractor/pandoratv.py

```diff
@@ -0,0 +1,40 @@
+# encoding: utf-8
+import re
+import json
+
+from .common import InfoExtractor
+from ..compat import (
+    compat_parse_qs,
+    compat_urllib_parse_urlparse,
+)
+from ..utils import (
+    find_xpath_attr,
+    unified_strdate,
+    get_element_by_attribute,
+    int_or_none,
+    qualities,
+)
+
+
+class PandoraTVIE(InfoExtractor):
+    _VALID_URL = r'http://channel.pandora.tv/channel/video.ptv\?ch_userid=(?P<userid>.*?)&prgid=(?P<prgid>.*?)&.*'
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('userid') + '_' + mobj.group('prgid')
+        webpage = self._download_webpage(url, video_id)
+        video_set = self._html_search_regex(
+                r'var videoSet = ({.*?});',
+                webpage, 'video set')
+        video_set = json.loads(video_set)
+        info_url ="http://channel.pandora.tv/channel/video.ptv?m=getSmartData&ch_userid={userid}&prgid={prgid}".format(**video_set)
+        info = self._download_webpage(info_url, None)
+        info = json.loads(info[1:-1])
```

Comment:
how would _download_json now that its a jsonp request that needs to be unwrapped(info[1:-1])?